samples/cxx/custom/custom.cpp

```diff
@@ -0,0 +1,197 @@
+
+        // pointer to the kinetics manager. provides iteration-specific species
+        // production rates based on the current state of the ThermoPhase.
+        kinetics = sol->kinetics();
+
+        // the system's constant pressure, taken from the provided initial state.
+        pressure = gas->pressure();
+
+        // number of chemical species in the system.
+        nSpecies = gas->nSpecies();
+
+        // resize the vector_fp storage containers for species partial molar enthalpies
+        // and net production rates. internal values are updated and used by the solver
+        // per iteration.
+        hbar.resize(nSpecies);
+        wdot.resize(nSpecies);
+
+        // number of equations in the ODE system. a conservation equation for each
+        // species, plus a single energy conservation equation for the system.
+        nEqs = nSpecies + 1;
+    }
+
+    /**
+     * Evaluate the ODE right-hand-side function, ydot = f(t,y).
+     *   - overrided from FuncEval, called by the integrator during simulation.
+     * @param[in] t time.
+     * @param[in] y solution vector, length neq()
+     * @param[out] ydot rate of change of solution vector, length neq()
+     * @param[in] p sensitivity parameter vector, length nparams()
+     */
+    void eval(double t, double* y, double* ydot, double* p) {
+        // the solution vector *y* is [T, Y_1, Y_2, ... Y_K], where T is the
+        // system temperature, and Y_k is the mass fraction of species k.
+        // similarly, the time derivative of the solution vector, *ydot*, is
+        // [dT/dt, Y_1/dt, Y_2/dt, ... Y_K/dt].
+        // the following variables are defined for clear and convenient access
+        // to these vectors:
+        double temperature = y[0];
+        double *massFracs = &y[1];
+        double *dTdt = &ydot[0];
+        double *dYdt = &ydot[1];
+        
+        /* ------------------------- UPDATE GAS STATE ------------------------- */
+        // the state of the ThermoPhase is updated to reflect the current solution
+        // vector, which was calculated by the integrator.
+        gas->setMassFractions_NoNorm(massFracs);
+        gas->setState_TP(temperature, pressure);
+
+        /* ----------------------- GET REQ'D PROPERTIES ----------------------- */
+        double rho = gas->density();
+        double cp = gas->cp_mass();
+        gas->getPartialMolarEnthalpies(&hbar[0]);
+        kinetics->getNetProductionRates(&wdot[0]);
+        
+        /* -------------------------- ENERGY EQUATION ------------------------- */
+        // the rate of change of the system temperature is found using the energy
+        // equation for a closed-system constant pressure ideal gas:
+        //     m*cp*dT/dt = - sum[h(k) * dm(k)/dt]
+        // or equivalently:
+        //     dT/dt = - sum[hbar(k) * dw(k)/dt] / (rho * cp)
+        double hdot_vol = 0;
+        for (int k = 0; k < nSpecies; k++)
+            hdot_vol += hbar[k] * wdot[k];
```

Comment:
I strongly recommend against using the unbraced (one line) `for` loop construct, here and elsewhere.
```suggestion
        for (int k = 0; k < nSpecies; k++) {
            hdot_vol += hbar[k] * wdot[k];
        }
```